File: airbyte-integrations/bases/base-java-s3/src/test/java/io/airbyte/integrations/destination/s3/avro/JsonFieldNameUpdaterTest.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3.avro;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.fasterxml.jackson.databind.JsonNode;
import io.airbyte.commons.json.Jsons;
import io.airbyte.commons.resources.MoreResources;
import io.airbyte.commons.util.MoreIterators;
import java.io.IOException;
import java.util.Map.Entry;
import java.util.stream.Collectors;
import org.junit.jupiter.api.Test;

class JsonFieldNameUpdaterTest {

  @Test
  public void testFieldNameUpdate() throws IOException {
    final JsonNode testCases = Jsons.deserialize(MoreResources.readResource("parquet/json_field_name_updater/test_case.json"));
    for (final JsonNode testCase : testCases) {
      final JsonNode nameMap = testCase.get("nameMap");
      final JsonFieldNameUpdater nameUpdater = new JsonFieldNameUpdater(
          MoreIterators.toList(nameMap.fields()).stream()
              .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().asText())));

      final JsonNode original = testCase.get("original");
      final JsonNode updated = testCase.get("updated");

      assertEquals(original, nameUpdater.getJsonWithOriginalFieldNames(updated));
    }
  }

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/bin/test/io/airbyte/cdk/integrations/destination/s3/avro/JsonFieldNameUpdaterTest.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.s3.avro

import io.airbyte.commons.json.Jsons.deserialize
import io.airbyte.commons.resources.MoreResources.readResource
import io.airbyte.commons.util.MoreIterators.toList
import java.io.IOException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class JsonFieldNameUpdaterTest {
    @Test
    @Throws(IOException::class)
    fun testFieldNameUpdate() {
        val testCases = deserialize(readResource("parquet/json_field_name_updater/test_case.json"))
        for (testCase in testCases) {
            val nameMap = testCase["nameMap"]
            val nameUpdater =
                JsonFieldNameUpdater(
                    toList(nameMap.fields()).associate { it.key to it.value.asText() }
                )

            val original = testCase["original"]
            val updated = testCase["updated"]

            Assertions.assertEquals(original, nameUpdater.getJsonWithOriginalFieldNames(updated))
        }
    }
}
